src/timeMachine/components/view_options/LegendOrientation.tsx

```diff
@@ -0,0 +1,123 @@
+// Libraries
+import React, {ChangeEvent, FC, useState} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
+import {get} from 'lodash'
+
+// Utils
+import {convertUserInputToNumOrNaN} from 'src/shared/utils/convertUserInput'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {getActiveTimeMachine} from 'src/timeMachine/selectors'
+
+// Components
+import {
+  ComponentStatus,
+  Form,
+  Grid,
+  Input,
+  InputType,
+  RangeSlider,
+} from '@influxdata/clockface'
+
+// Types
+import {AppState} from 'src/types'
+
+// Constants
+import {
+  LEGEND_OPACITY_DEFAULT,
+  LEGEND_OPACITY_MAXIMUM,
+  LEGEND_OPACITY_MINIMUM,
+  LEGEND_OPACITY_STEP,
+} from 'src/shared/constants'
+
+interface OwnProps {
+  onLegendOpacityChange: (opacity: number) => void
+  onLegendOrientationThresholdChange: (threshold: number) => void
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = OwnProps & ReduxProps
+
+const LegendOrientation: FC<Props> = props => {
+  const {
+    legendOpacity,
+    onLegendOpacityChange,
+    legendOrientationThreshold,
+    onLegendOrientationThresholdChange,
+  } = props
+
+  const [thresholdInputStatus, setThresholdInputStatus] = useState(
+    ComponentStatus.Default
+  )
+  const [thresholdInput, setThresholdInput] = useState(
+    legendOrientationThreshold
+  )
+
+  const hanldeSetThreshold = (event: ChangeEvent<HTMLInputElement>): void => {
+    const value = convertUserInputToNumOrNaN(event)
+
+    setThresholdInput(value)
+    if (isNaN(value)) {
+      setThresholdInputStatus(ComponentStatus.Error)
+    } else {
+      setThresholdInputStatus(ComponentStatus.Default)
+      onLegendOrientationThresholdChange(value)
+    }
+  }
+
+  const handleSetOpacity = (event: ChangeEvent<HTMLInputElement>): void => {
+    const value = convertUserInputToNumOrNaN(event)
+
+    if (isNaN(value) || value < LEGEND_OPACITY_MINIMUM) {
+      onLegendOpacityChange(LEGEND_OPACITY_MAXIMUM)
+    } else {
+      onLegendOpacityChange(value)
+    }
+  }
+
+  if (!isFlagEnabled('legendOrientation')) {
+    return null
+  }
+  return (
+    <Grid.Column>
+      <h5 className="view-options--header">Legend</h5>
+      <Form.Element label="Horizontal Column Count Threshold">
```

Comment:
I'm tempted to label this `Rotation Threshold` and then add help text below saying `Legends with column count exceeding the threshold appear horizontally instead of vertically`

helpText: https://influxdata.github.io/clockface/?path=/story/components-forms-standard--formelement

Reply:
Yes, your suggestion is much better. I will make that change.